### PR TITLE
Revert "Add batchprocessor support for client metadata (#7578)"

### PR DESCRIPTION
**Pull request by bogdandrutu**
This reverts commit aa206e399cc02aa096d52f09efc07cbdc2275561.

Start/Stop logic is broken, see https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5048869164/jobs/9057623732

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
cc @jmacd

File: .chloggen/revertbatch.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: batchprocessor
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Revert "Add batchprocessor support for client metadata (#7578)"
```
Comment (jpkrohling) [inline .chloggen/revertbatch.yaml:8]: If this hasn't been released yet, shouldn't the revert just remove the changelog entry for the thing being reverted?

Comment (dmitryax) [reply .chloggen/revertbatch.yaml:8]: 0.78.0 is release

Comment (jpkrohling) [reply .chloggen/revertbatch.yaml:8]: time flies -- I thought this was added after the last release

**Comment by atoulme** [comment]
With the fix for start/stop merged, is this PR still in need to be merged?

**Pull request closed without merging**